main.py

```diff
@@ -357,15 +363,27 @@ async def on_message(message: discord.Message):
         await bot.process_commands(message)
         return
 
+    message_copy = MessageCopy(message.content, message.author.display_name, message.author.avatar_url)
+
     LOGGER.info("Beginning message listener stack execution.")
     for listener in message_listeners:
         LOGGER.info(f"Executing: {listener}")
-        if await listener(message):  # Return early if any listeners return true.
-            return
+
+        # If a message listener accepts two parameters, that means it is able to transform the message copy.
+        # This means it affects a user's display name, avatar, or message content. Such as speech optimization or identity enforcement.
+
+        if len(signature(listener).parameters) == 2:
+            if await listener(message, message_copy):  # Return early if any listeners return true.
+                return
+        else:
+            if await listener(message):  # Return early if any listeners return true.
+                return
```

Comment:
I'm not too happy about using reflection like this. I understand that the alternative is a very time-intensive rewrite of all listeners, so it might be okay as a temporary solution. But within the next few weeks I want this to be replaced with just:
```py
if await listener(message, message_copy):  # Return early if any listeners return true.
            return
```